### ISSUE TITLE
References do not appear in PDF outline

### ISSUE THREAD

**Issue by jack-jjm** [typst/typst#5573]
### Description

If you create a section heading which includes a reference in the title, the reference does not appear in the PDF outline. For example:

```
== Full proof of @theorem:equivalence
```

In the PDF outline, this will look something like this:

![image](https://github.com/user-attachments/assets/9dac675d-2a3f-40ed-8201-73713c3e377a)

Typst 0.12

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enivex** [comment]
I cannot reproduce this. Can you provide an MVE?

![image](https://github.com/user-attachments/assets/79c836f5-fb5e-40af-8a03-ed03cb227a53)

**Comment by laurmaedje** [comment]
Yeah, it's a known limitation. The plain text conversion used for the outline is very primtive.

**Comment by laurmaedje** [comment]
Sort of related: https://github.com/typst/typst/issues/2415

**Comment by laurmaedje** [comment]
Yeah, that's also related
